crates/agent/src/modules/wallet.rs

```diff
@@ -0,0 +1,108 @@
+use crate::error::Result;
+use async_trait::async_trait;
+use serde::{Deserialize, Serialize};
+
+/// Options that are supplied when querying a wallet for DIDs
+#[derive(Debug, Deserialize, Serialize)]
+pub struct Did {
+    /// The DID of interest
+    pub did: Option<String>,
+
+    // TODO: enum
+    /// The key type to query for eg. ed25519, bls1238g2
+    pub key_type: Option<String>,
+
+    // TODO: enum
+    /// DID method to query for. e.g. sov to only fetch indy/sov DIDs Available values : key, sov
+    pub method: Option<String>,
+
+    // TODO: enum
+    /// Whether DID is current public DID, posted to ledger but current public DID, or local to the wallet
+    /// Available values : public, posted, wallet_only
+    pub posture: Option<String>,
+
+    /// The verification key of interest
+    pub verkey: Option<String>,
+}
+
+/// Response from the cloudagent when requesting info about dids
+/// of a wallet
+#[derive(Debug, Deserialize, Serialize)]
+pub struct DidList {
+    /// List of all the ids of every schema that the cloudagent has registered
+    pub results: Vec<Did>,
+}
```

Comment:
It is a difficult thing, but the whole API (this also in particular) is quite ACA-Py specific. We can keep it in, but we have to change some things here.

I think we could do something like this (and fix the result in the aca-py):

```suggestion
#[derive(Debug, Deserialize, Serialize)]
pub struct DidList(Vec<Did>);
```